src/transforms/metric_to_log.rs

```diff
@@ -0,0 +1,346 @@
+use super::Transform;
+use crate::{
+    config::{log_schema, DataType, TransformConfig, TransformContext, TransformDescription},
+    event::{self, Event, LogEvent},
+    internal_events::{
+        MetricToLogEventProcessed, MetricToLogFailedDeserialize, MetricToLogFailedSerialize,
+    },
+    types::Conversion,
+};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct MetricToLogConfig {
+    pub host_tag: Option<Atom>,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<MetricToLogConfig>("metric_to_log")
+}
+
+#[typetag::serde(name = "metric_to_log")]
+impl TransformConfig for MetricToLogConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(MetricToLog::new(self.host_tag.clone())))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Metric
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "metric_to_log"
+    }
+}
+
+pub struct MetricToLog {
+    timestamp_key: Atom,
+    host_tag: Atom,
+}
+
+impl MetricToLog {
+    pub fn new(host_tag: Option<Atom>) -> Self {
+        Self {
+            timestamp_key: Atom::from("timestamp"),
+            host_tag: Atom::from(format!(
+                "tags.{}",
+                host_tag.unwrap_or_else(|| log_schema().host_key().clone())
+            )),
+        }
+    }
+}
+
+impl Transform for MetricToLog {
+    fn transform(&mut self, event: Event) -> Option<Event> {
+        let metric = event.into_metric();
+        emit!(MetricToLogEventProcessed);
+
+        serde_json::to_vec(&metric)
```

Comment:
Instead of going all the way to bytes, it'd probably be faster to use `to_value`.